os/arch/arm/src/common/up_mpuinit.c

```diff
@@ -50,11 +51,102 @@
 /****************************************************************************
  * Pre-processor Definitions
  ****************************************************************************/
+static struct mpu_usages_s g_mpu_region_usages;
+
+#ifdef CONFIG_ARMV7M_MPU
+#define MPU_MAX_NREGION		CONFIG_ARMV7M_MPU_NREGIONS
+#elif defined(CONFIG_ARMV8M_MPU)
+#define MPU_MAX_NREGION		CONFIG_ARMV8M_MPU_NREGIONS
+#endif
 
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
 
+/****************************************************************************
+ * Name: mpu_regs_initialize
+ *
+ * Description:
+ *   Initialize the MPU region numbers.
+ *
+ ****************************************************************************/
+
+void mpu_region_initialize(struct mpu_usages_s *mpu)
+{
+	uint8_t offset = (struct mpu_usages_s *)mpu->nregion_board_specific;
+
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
+	offset += MPU_NUM_REGIONS;
+#else
+	offset += MPU_NUM_REGIONS;
+#endif
```

Comment:
#ifdef and #else are the same. Let's remove the conditional and duplicated line.